data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/GenericExpressionEvaluator.java

```diff
@@ -52,4 +52,27 @@ public Boolean isValidExpressionStatement(final String statement) {
             return false;
         }
     }
+
+    @Override
+    public Boolean isValidFormatExpressions(final String format) {
+        int fromIndex = 0;
+        int position = 0;
+        while ((position = format.indexOf("${", fromIndex)) != -1) {
+            int endPosition = format.indexOf("}", position + 1);
+            if (endPosition == -1) {
+                return false;
+            }
+            String name = format.substring(position + 2, endPosition);
+
+            Object val;
+            // We only check the expression if it matches (.*) to mitigate the issue with the antlr logger ()
+            // These can't be keys because (, ) is invalid for keys, so we attempt to validate the expression. All expressions currently
+            // contain a function ( ) so this will detect them to validate against.
+            if (name.matches("(.*)") && !isValidExpressionStatement(name)) {
```

Comment:
Isn't this (doing `matches()`) a very expensive operation to do every time? Is it just making sure there is `(` in the beginning and `)` at the end?

Reply:
This validation is only supposed to be called to validate the format expression on startup. It's a way to avoid running `isValidExpressionStatement` and to have those antlr logs show up and to let it be valid when the usage is `${key}`. The `()` is meant to check that we are using an expression function, so we call validate. I'm not sure of the best way to do this.

Reply:
Why not just check for first character being '(' and last character being ')'. Doing regex match is lot more expensive if we are doing once. You should do something like
```
if (name.charAt(0) == '(' && name.charAt(name.length()-1) == ')' && !isValidExpressionStatement(name))
```

Reply:
It doesn't start with `(` though? Does it always end with `)`? I could change it to just check for the ending with `)`

Reply:
Changed to check if it's a valid key or a valid expression, and only is invalid when both are the case. This covers a lot since `(` is an invalid character in keys